backstory/src/main/java/com/google/sps/story/PromptManagerBodyFactory.java

```diff
@@ -0,0 +1,228 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.sps.story;
+
+import com.google.sps.story.data.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Scanner;
+
+/**
+ * Constructs body of prompt given input keywords.
+ */
+public class PromptManagerBodyFactory {
+  /** Keywords for prompt generation */
+  private List<String> keywords;
+  /** Word processing client */
+  private PromptManagerWordTools wordTools;
+  /** randomness flag */
+  private boolean chooseRandomly;
+
+  /** Random number generator */
+  private Random templateChooser;
+
+  /** Data structures to hold classified words */
+  private List<String> gerunds;
+  private List<String> nouns;
+  private Map<WordType, List<String>> typeMap;
+
+  /** Minimum number of nouns for descriptive method. */
+  private final int MINIMUM_NOUNS = 3;
+
+  /** Final output string. */
+  private String outputBody;
+
+  /** Templates for introductory sentence completion */
+  private final String[] INTRO_TEMPLATES = {
+      "a <adj> <noun> as well as a <adj> <noun> decided to come together.",
+      "a <adj> <adj> <noun> and a <adj> <adj> <noun> appeared all at once. "};
+
+  /** Templates for introductory sentence completion with gerunds included */
+  private final String[] INTRO_TEMPLATES_WITH_GERUNDS = {
+      "there was a <adj> <adj> <noun> <gerund> alongside a <adj> <noun>.",
+      "a <adj> <noun> as well as a <gerund> <adj> <noun> were together."};
+
+  /** Templates for second sentence. */
+  private final String[] SECOND_TEMPLATES = {
+      "A <adj> <noun> was also present, quite an interesting scene.",
+      "One must not forget the <adj> <noun>, which simply cannot be ignored.",
+  };
+
+  /** Templates for list method sentence endings. */
+  private final String[] SIMPLE_ENDINGS = {"were all really quite interesting.",
+      "all came together in one place.", "were all together at once."};
+
+  /** Default ending for empty string return. */
+  private final String EMPTY_ENDING = "a hectic, unrecognizable scene took place.";
+
+  /**
+   * Initializes fields and takes in word processing client if supplied.
+   *
+   * @param keywords Input words for generation.
+   * @param chooseRandomly Whether or not to randomly select a template.
+   * @param wordTools Client for word processing API calls.
+   */
+  public PromptManagerBodyFactory(
+      List<String> keywords, boolean chooseRandomly, PromptManagerWordTools wordTools) {
+    this(keywords, chooseRandomly);
+    this.wordTools = wordTools;
+  }
+
+  /**
+   * Initializes fields.
+   *
+   * @param keywords Input words for generation.
+   * @param chooseRandomly Whether or not to randomly select a template.
+   */
+  public PromptManagerBodyFactory(List<String> keywords, boolean chooseRandomly) {
```

Comment:
This name sounds like it's a factory of PromptManagerBody instances, i.e. this would be expected:

```
public PromptManagerBodyFactory {
  PromptManagerBody newInstance();
}
```

So I don't think this is really a factory... maybe `Generator`?

```
PromptManagerBodyGenerator() {
  String generate();
}
```

Reply:
Done